### ISSUE TITLE
OmniSharp doesn't work for WebApplication projects

### ISSUE THREAD

**Issue by m0sa** [OmniSharp/omnisharp-vscode#1368]
## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (1.0.2)

Product Information:
 Version:            1.0.2
 Commit SHA-1 hash:  54743c8b7b

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.2
```
VS Code version: 1.12.0 insiders
C# Extension version: 1.8.1

![image](https://cloud.githubusercontent.com/assets/503702/24789904/36118e7e-1b75-11e7-9a3a-2d501468e5c2.png)

## Steps to reproduce
Open a folder with an WebApplication csproj in vscode

## Expected  behavior
Omnisharp should work

## Actual behavior
Omnisharp doesn't load the csproj, the relevant errors in the output:

```
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Loading project from 'SomeProject.csproj'.
[warn]: OmniSharp.MSBuild.MSBuildProjectSystem
        Failed to process project file 'SomeProject.csproj'.
SomeProject.csproj(1,1)
Microsoft.Build.Exceptions.InvalidProjectFileException: The imported project "~\.vscode-insiders\extensions\ms-vscode.csharp-1.8.1\bin\omnisharp\msbuild\Microsoft\VisualStudio\v15.0\WebApplications\Microsoft.WebApplication.targets" was not found. Also, tried to find "WebApplications\Microsoft.WebApplication.targets" in the fallback search path(s) for $(VSToolsPath) - "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v15.0" . These search paths are defined in "~\.vscode-insiders\extensions\ms-vscode.csharp-1.8.1\bin\omnisharp\OmniSharp.exe.Config". Confirm that the path in the <Import> declaration is correct, and that the file exists on disk in one of the search paths.  SomeProject.csproj
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
   at Microsoft.Build.Evaluation.Evaluator`4.ThrowForImportedProjectWithSearchPathsNotFound(ProjectImportPathMatch searchPathMatch, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImports(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate()
   at Microsoft.Build.Evaluation.Project.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings)
   at Microsoft.Build.Evaluation.Project.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings)
   at Microsoft.Build.Evaluation.Project.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings)
   at Microsoft.Build.Evaluation.Project..ctor(String projectFile, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings)
   at Microsoft.Build.Evaluation.ProjectCollection.LoadProject(String fileName, IDictionary`2 globalProperties, String toolsVersion)
   at OmniSharp.MSBuild.ProjectFile.ProjectFileInfo.Create(String projectFilePath, String solutionDirectory, ILogger logger, MSBuildOptions options, ICollection`1 diagnostics, Boolean isUnityProject)
   at OmniSharp.MSBuild.MSBuildProjectSystem.CreateProjectFileInfo(String projectFilePath, Boolean isUnityProject)
```

The relevant line from the .csproj file:

```
<Import Project="$(VSToolsPath)\WebApplications\Microsoft.WebApplication.targets" Condition="'$(VSToolsPath)' != ''" />
```

**Comment by DustinCampbell** [comment]
OmniSharp doesn't include all of the necessary tasks and targets needed to process any project. The error that MSBuild returned is telling:

> ```The imported project "~\.vscode-insiders\extensions\ms-vscode.csharp-1.8.1\bin\omnisharp\msbuild\Microsoft\VisualStudio\v15.0\WebApplications\Microsoft.WebApplication.targets" was not found. Also, tried to find "WebApplications\Microsoft.WebApplication.targets" in the fallback search path(s) for $(VSToolsPath) - "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v15.0" . These search paths are defined in "~\.vscode-insiders\extensions\ms-vscode.csharp-1.8.1\bin\omnisharp\OmniSharp.exe.Config". Confirm that the path in the <Import> declaration is correct, and that the file exists on disk in one of the search paths.  SomeProject.csproj```

Do you have Microsoft.WebApplication.targets anywhere on your system?

**Comment by m0sa** [comment]
Yes the file is in `C:\Program Files (x86)\Microsoft Visual
Studio\2017\Professional\MSBuild\Microsoft\VisualStudio\v15.0\WebApplications`,
the build works in VS2017, it worked in previous versions of
vscode/omnisharp.

Samo

2017-04-07 17:24 GMT+02:00 Dustin Campbell <notifications@github.com>:

**Comment by DustinCampbell** [comment]
When was it working? Did it work recently in C# for VS Code but some update (maybe to VS 2017) broke it?

**Comment by DustinCampbell** [comment]
I tried creating a new ASP .NET Web Application (MVC) in VS 2017 and didn't encounter this error. Do you have project that you'd be willing to share out to help repro this problem?

**Comment by DustinCampbell** [comment]
Eureka! I can repro if I launch VS Code from a VS 2017 Developer Command Prompt, but not from a normal Windows Command Prompt. Do you see the same behavior?

**Comment by DustinCampbell** [comment]
OK. Digging in a bit, it seems that the presence of the `VisualStudioVersion=15.0` environment variable causes it to fail. I suspect this is causing MSBuild to take some other code path. Digging further...

**Comment by m0sa** [comment]
Yup, I can confirm it only happens in a 2017 Developer Command Prompt. Thanks for digging into it.

**Comment by DustinCampbell** [comment]
This will be fixed when we take a new build of OmniSharp into the VS Code extension.

**Comment by darkguy2008** [comment]
We're almost in June, when will that be?

**Comment by DustinCampbell** [comment]
This already happened and the issue was closed. It was fixed in C# for VS Code 1.9. if you're experiencing a similar problem, could you file a new issue?
